packages/core/src/commands/setContent.ts

```diff
@@ -58,6 +58,10 @@ export const setContent: RawCommands['setContent'] = (content, emitUpdate = fals
 
     if (dispatch) {
       tr.replaceWith(0, doc.content.size, document).setMeta('preventUpdate', !emitUpdate)
+
+      Object.entries(document.attrs).forEach(([key, value]) => {
+        tr.setDocAttribute(key, value)
+      })
```

Comment:
We are planning to remove this branch in the next version (and already have).

So, I don't think that this is the right place for it.

I'm not sure that a method named `setContent` _should_ update the doc's attributes. I think you may want to implement this in your application instead. I don't think it belongs in this command

Reply:
We tend not to promote doc attributes because they are not synchronized in collaboration with y-prosemirror because of a long standing bug.

So, we don't want to promote something that would not work in the Tiptap ecosystem

Reply:
Alright, I do understand.